core/src/repair/quic_endpoint.rs

```diff
@@ -0,0 +1,624 @@
+#![allow(dead_code)]
+use {
```

Comment:
I have not examined this in detail. But I think you are using the bi-directional streams? This will requires duplications of the existing streamer/quic code. I think a much simpler way is keeping using the existing streamer/quic code using uni streams. On both the client and server side, we setup the streamer (for accepting requests and accepting responses) pretty similar to how the way using UDP transport is done. This will introduce minimal changes and take advantage of existing code.

Reply:
This was very intentional.

Current repair code is using [nonce](https://github.com/solana-labs/solana/blob/e6dc3dac4/core/src/repair/serve_repair.rs#L198), [outstanding-requests](https://github.com/solana-labs/solana/blob/e6dc3dac4/core/src/repair/outstanding_requests.rs), [request_response](https://github.com/solana-labs/solana/blob/e6dc3dac4/core/src/repair/request_response.rs), [signature verification](https://github.com/solana-labs/solana/blob/e6dc3dac4/core/src/repair/serve_repair.rs#L843-L896), `num_expected_responses` etc to partially and imperfectly re-invent QUIC over UDP.

With bidirectional streams we get these by default plus much better control on the security aspects of repair.

> pretty similar to how the way using UDP transport is done.

UDP transport is one-directional; repair is definitely a request-response model, with security implications on the response (hence the reason for nonce, outstanding-requests, etc).
With UDP going away, rather than sticking to UDP model, this can be implemented in much cleaner and secure way using QUIC bi-directional streams.